### PR TITLE
Upgrade to Django 4.0

### PR DESCRIPTION
**Pull request by ghickman**
This upgrades us to Django 4, fixing the URL tests to be more correct along the way.

### CODE REVIEW

File: jobserver/settings.py [hunk 1]
```diff
@@ -146,7 +146,6 @@
 LANGUAGE_CODE = "en-us"
 TIME_ZONE = "UTC"
 USE_I18N = True
-USE_L10N = True
```
Comment (StevenMaude) [inline jobserver/settings.py:149]: note: This is now a default, so no change in behaviour.

File: tests/unit/jobserver/test_urls.py [hunk 2]
```diff
@@ -145,9 +145,17 @@ def test_url_redirects(client, url, redirect):
 )
 def test_url_resolution(url, view):
     """Test each URL resolves to the expected view function or class"""
-    resolved_view = resolve(url).func
+    match = resolve(url)
 
+    # get the actual class for a CBV.  View.as_view() constructs a new wrapper
+    # function to be the callable which url resolution calls when a view is
+    # accessed.  We don't want that callalbe at this point in the test because
```
Comment (StevenMaude) [inline tests/unit/jobserver/test_urls.py:152]: nit: `callalbe` typo

File: requirements.prod.txt [hunk 3]
```diff
@@ -402,6 +402,10 @@ python3-openid==3.2.0 \
     --hash=sha256:33fbf6928f401e0b790151ed2b5290b02545e8775f982485205a066f874aaeaf \
     --hash=sha256:6626f771e0417486701e0b4daff762e7212e820ca5b29fcc0d05f6f8736dfa6b
     # via social-auth-core
+pytz==2021.3 \
```
Comment (StevenMaude) [inline requirements.prod.txt:405]: issue: aeb47d24203cff9bd5f9911bb392b02ffc2429dc states:

> Upgrade DRF 3.12.4 -> 3.13.1 to drop pytz dependency

But `pytz` is still here, so either the commit message is not accurate, or the requirements need rebuilding.

**Review by StevenMaude — APPROVED** [review]
summary: :+1: there's a couple of small comments to look at, but overall this is good, so approving.

File: requirements.prod.txt (hunk 3 above)
Comment (ghickman) [reply requirements.prod.txt:438]: Arg, they pulled it into their requirements rather than removing the dep, thanks 👍

**Pull request merged**